### PR TITLE
Fix `censorCoordinates` recalc bug

### PR DESCRIPTION
**Pull request by ZeroMemes**
See #3565

From my own testing (by going through each usage of `FORCE_REVALIDATE_GOAL_AND_PATH` manually), everything should be correct

### CODE REVIEW

**Comment by leijurv** [comment]
brady hates subclassing? he uses `getClass` instead of `instanceof`??

**Comment by ZeroMemes** [comment]
IntelliJ automatic equals generation LOl

**Comment by leijurv** [comment]
also le baritone style is `{}` around all `if`s 😤

**Comment by ZeroMemes** [comment]
Can't have `new GoalBreak(BlockPos.ZERO).equals(new GoalGetToBlock(BlockPos.ZERO)) == true` 😤

**Review by ZacSharp — APPROVED** [review]

File: src/main/java/baritone/process/MineProcess.java [hunk 1]
```diff
@@ -319,6 +319,21 @@ public double heuristic(int x, int y, int z) {
             int zDiff = z - this.z;
             return GoalBlock.calculate(xDiff, yDiff < -1 ? yDiff + 2 : yDiff == -1 ? 0 : yDiff, zDiff);
         }
+
+        @Override
+        public boolean equals(Object o) {
+            return super.equals(o);
+        }
```
Comment (ZacSharp) [inline src/main/java/baritone/process/MineProcess.java:326]: Why no early return for mismatched classes here?
A `GoalThreeBlocks` and a `GoalTwoBlocks` don't have the same heuristic or goal condition so they aren't equals.

**Pull request merged**